index.js

```diff
@@ -1092,7 +1092,9 @@ Primus.prototype.reserved.events = {
  * @api public
  */
 Primus.createSocket = function createSocket(options) {
-  options = options || {};
+  options = Primus.prototype.merge.call(Primus, {}, options || {});
```

Comment:
It's used in the server code of the transformers (e.g. https://github.com/primus/primus/blob/509cf0294728a5b2000b825d40c56560b0f4e5c6/transformers/websockets/server.js#L15) so feel free to use it.